Makefile

```diff
@@ -0,0 +1,60 @@
+PROJECT := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+SRC := $(PROJECT)src
+TESTS := $(PROJECT)tests
+ALL := $(SRC) $(TESTS)
+
+export PYTHONPATH = $(PROJECT):$(PROJECT)/lib:$(SRC)
+export PY_COLORS=1
+
+# Update uv.lock with the latest deps
+lock:
+	uv lock --upgrade --no-cache
+
+# Generate requirements.txt from pyproject.toml
+requirements:
+	uv export --frozen --no-hashes --format=requirements-txt -o requirements.txt
```

Comment:
Have we figured out why output redirection didn't work in the `charmcraft pack` context? I'm concerned that might *make* us lose some time in the future

Reply:
No solution yet ...